app/components/Tests/Lozenge.test.tsx

```diff
@@ -0,0 +1,108 @@
+import { render, fireEvent } from "@testing-library/react-native"
+import { TamaguiProvider, Theme } from "tamagui"
+
+import { ThemeProvider } from "@/theme/context"
+
+import config from "../../tamagui.config"
+import { Lozenge } from "../Lozenge"
+
+const Providers: React.FC<React.PropsWithChildren> = ({ children }) => (
+  <TamaguiProvider config={config}>
+    <Theme name="light">
+      <ThemeProvider>{children}</ThemeProvider>
+    </Theme>
+  </TamaguiProvider>
+)
+
+const renderWithProviders = (ui: React.ReactElement) => {
+  return render(ui, { wrapper: Providers })
+}
+
+describe("Lozenge", () => {
+  it("renders correct text for each type", () => {
+    const { getByText } = renderWithProviders(<Lozenge type="leave" active />)
+    expect(getByText("Leave")).toBeTruthy()
+
+    const { getByText: getByText2 } = renderWithProviders(<Lozenge type="swap" active />)
+    expect(getByText2("Swap")).toBeTruthy()
+
+    const { getByText: getByText3 } = renderWithProviders(<Lozenge type="event" />)
+    expect(getByText3("Event")).toBeTruthy()
+
+    const { getByText: getByText4 } = renderWithProviders(<Lozenge type="available" />)
+    expect(getByText4("Available")).toBeTruthy()
+
+    const { getByText: getByText5 } = renderWithProviders(<Lozenge type="requested" />)
+    expect(getByText5("Requested")).toBeTruthy()
+
+    const { getByText: getByText6 } = renderWithProviders(<Lozenge type="urgent" />)
+    expect(getByText6("Urgent")).toBeTruthy()
+
+    const { getByText: getByText7 } = renderWithProviders(<Lozenge type="approved" />)
+    expect(getByText7("Approved")).toBeTruthy()
+
+    const { getByText: getByText8 } = renderWithProviders(<Lozenge type="awaiting" />)
+    expect(getByText8("Awaiting")).toBeTruthy()
+
+    const { getByText: getByText9 } = renderWithProviders(<Lozenge type="declined" />)
+    expect(getByText9("Declined")).toBeTruthy()
+
+    const { getByText: getByText10 } = renderWithProviders(<Lozenge type="assigned" />)
+    expect(getByText10("Assigned")).toBeTruthy()
+
+    const { getByText: getByText11 } = renderWithProviders(<Lozenge type="openShift" />)
+    expect(getByText11("OpenShift")).toBeTruthy()
+  })
```

Comment:
Let's not repeat ourselves.

Separate out and export LOZENGE_CONFIG in the `Lozenge.tsx` file like this:
```
export const LOZENGE_CONFIG: Record<LozengeType, LozengeConfig> = {
```

Then use it here like this:


```suggestion
it("renders correct text for each type", () => {
  Object.keys(LOZENGE_CONFIG).forEach((type) => {
    const { getByText } = renderWithProviders(<Lozenge type={type as any} active />)
    const expectedText = LOZENGE_CONFIG[type as keyof typeof LOZENGE_CONFIG].text
    expect(getByText(expectedText)).toBeTruthy()
  })
})
```